### PR TITLE
gentool accept config file #252

### PR DESCRIPTION
**Pull request by dino-ma**
- [X] Do only one thing
- [X] Non breaking API changes
- [X] Tested

### What did this pull request do?

Replace the command line with a configuration file.
Add gen.yml file.
Add command c is default blank.

### User Case Description

 gentool -c "gen.yml"

### CODE REVIEW

File: tools/gentool/gentool.go [hunk 1]
```diff
@@ -4,8 +4,10 @@ import (
 	"flag"
 	"fmt"
 	"log"
+	"os"
 	"strings"
 
+	"gopkg.in/yaml.v2"
```
Comment (riverchu) [inline tools/gentool/gentool.go:10]: Why not using `yaml.v3`?

File: tools/gentool/gentool.go [hunk 2]
```diff
@@ -61,9 +85,26 @@ func getModels(g *gen.Generator, db *gorm.DB, tables []string) (models []interfa
...
 
-func main() {
+// loadConfigFile load config file from path
+func loadConfigFile(path string) (*CmdParams, error) {
+	file, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+	var yamlConfig *YamlConfig
```
Comment (riverchu) [inline tools/gentool/gentool.go:95]: Maybe create struct directly is better?
```go
var yamlConfig YamlConfig
```

Comment (dino-ma) [reply tools/gentool/gentool.go:95]: yet

File: tools/gentool/gentool.go (hunk 1 above)
Comment (dino-ma) [reply tools/gentool/gentool.go:10]: I think you use 2, so you keep it.
Let me update it.

File: tools/gentool/gentool.go [hunk 3]
```diff
@@ -73,25 +114,49 @@ func main() {
 	fieldWithIndexTag := flag.Bool("fieldWithIndexTag", false, "generate field with gorm index tag")
 	fieldWithTypeTag := flag.Bool("fieldWithTypeTag", false, "generate field with gorm column type tag")
 	flag.Parse()
-
-	db, err := connectDB(DBType(*dbType), *dsn)
-	if err != nil {
-		log.Fatalln("connect db server fail:", err)
+	if *genPath != "" {
+		return loadConfigFile(*genPath)
```
Comment (riverchu) [inline tools/gentool/gentool.go:118]: return config here will lead to lossing information which specified directly in command line arguments

**Comment by dino-ma** [comment]
done @riverchu

**Pull request merged**